service/src/main/kotlin/no/nav/su/se/bakover/service/dokument/DokumentServiceImpl.kt

```diff
@@ -0,0 +1,172 @@
+package no.nav.su.se.bakover.service.dokument
+
+import arrow.core.Either
+import arrow.core.getOrElse
+import arrow.core.left
+import arrow.core.right
+import no.nav.su.se.bakover.client.dokarkiv.DokArkiv
+import no.nav.su.se.bakover.client.dokarkiv.Journalpost
+import no.nav.su.se.bakover.client.dokarkiv.JournalpostFactory
+import no.nav.su.se.bakover.client.dokarkiv.JournalpostSkatt.Companion.lagJournalpost
+import no.nav.su.se.bakover.client.dokdistfordeling.DokDistFordeling
+import no.nav.su.se.bakover.common.journal.JournalpostId
+import no.nav.su.se.bakover.domain.brev.BrevbestillingId
+import no.nav.su.se.bakover.domain.brev.Distribusjonstidspunkt
+import no.nav.su.se.bakover.domain.brev.Distribusjonstype
+import no.nav.su.se.bakover.domain.brev.KunneIkkeBestilleBrevForDokument
+import no.nav.su.se.bakover.domain.brev.KunneIkkeDistribuereBrev
+import no.nav.su.se.bakover.domain.brev.KunneIkkeJournalføreBrev
+import no.nav.su.se.bakover.domain.brev.KunneIkkeJournalføreDokument
+import no.nav.su.se.bakover.domain.dokument.DokumentRepo
+import no.nav.su.se.bakover.domain.dokument.Dokumentdistribusjon
+import no.nav.su.se.bakover.domain.eksterneiverksettingssteg.KunneIkkeJournalføreOgDistribuereBrev
+import no.nav.su.se.bakover.domain.person.PersonService
+import no.nav.su.se.bakover.domain.sak.SakService
+import no.nav.su.se.bakover.domain.skatt.DokumentSkattRepo
+import no.nav.su.se.bakover.domain.skatt.Skattedokument
+import no.nav.su.se.bakover.service.dokument.DokumentResultatSet.Companion.logResultat
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class DokumentServiceImpl(
+    private val dokArkiv: DokArkiv,
+    private val dokDistFordeling: DokDistFordeling,
+    private val sakService: SakService,
+    private val personService: PersonService,
+    private val dokumentRepo: DokumentRepo,
+    private val dokumentSkattRepo: DokumentSkattRepo,
+) : DokumentService {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+
+    override fun journalførDokumenter() {
+        val dokumenterSomMåJournalføres = dokumentRepo.hentDokumenterForJournalføring()
+        val skatteDokumenterSomMåJournalføres = dokumentSkattRepo.hentDokumenterForJournalføring()
+
+        val dokumenterResultat = dokumenterSomMåJournalføres.map { dokumentdistribusjon ->
```

Comment:
Hadde vært fint og delt opp dette i 2 funksjoner, en for vanlig dokumenter og en for skatt?